### ISSUE TITLE
Insane/Expensive import time

### ISSUE THREAD

**Issue by jesusch** [awslabs/aws-data-wrangler#460]
Given the following AWS Lambda function
```
import time

def lambda_handler(event, context):
    start = time.time()
    import awswrangler
    print('import time: %f' % (time.time() - start))
```

Resulting import time (no - not Init Duration):
128MB RAM -> 18.847529 s.  -> .0000039259 $
1024MB RAM -> 2.834644 s. -> .0000047245 $
3008MB RAM -> 1.223252 s. -> .0000059887 $

Used environment:
python3.8
awswrangler==1.10.0 via AWSLayer
Prices according to Ohio

that's a very expensive import!

**Comment by jesusch** [comment]
Output from 
`python -X importtime -c 'import awswrangler'`
https://gist.github.com/jesusch/318349f230dd25233b19793fb7301bd3

**Comment by igorborgest** [comment]
Hi @jesusch! Thanks for reaching out!

### 1 - Heavy dependencies

AWS Data Wrangler relies on several heavy dependencies. Just the `Numpy+Pandas+PyArrow` combo will consume something around 0.5 second with the biggest Lambda config. In addition, we also have `boto3+botocore` and `sqlalchemy` that also have relative big startup time.

### 2 - Import all at once by design

Wrangler's modules are highly interconnected, so even if we break it in some separated sub packages, we would see a marginal speed improvement trying to import only some specific group of functionalities. 

In addition, by design Wrangler aims to be user friendly with all functionalities under the `wr` alias (`import awswrangler as wr`). It allows full autocomplete and easier feature discovery.

### 3 - Lambda CPU limitation

Unfortunately 128MB and 1024MB are really small, it will run with a fraction of CPU, so it is not even a whole core available for you. I would say 2048MB is the minimum configuration to handle it properly.

### 4 - Lambda execution environment

The lambda execution environment will help you to skip this import time if your Lambda is warmed. So you probably only will "pay" for this initialization time on cold starts.

> Objects declared outside of the function's handler method remain initialized, providing additional optimization when the function is invoked again. For example, if your Lambda function establishes a database connection, instead of reestablishing the connection, the original connection is used in subsequent invocations. We recommend adding logic in your code to check if a connection exists before creating a new one.

[Reference](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-context.html)

### 5 - General recommendations

In the end, we will always try to optimize the library, but AWS Data Wrangler will always have some import footprint regarding its heavy dependencies. So I recommend you to always run with at least 1 full CPU and to try to optimize your workload to "pay" less times this import cost (E.g. If your files is too small you should consider to process more than 1 file per lambda execution).

### 6 - Some improvements in the commit above (~30%)

Trying to find some opportunities to improve this import time, I found out that `sqlalchemy_redshift` consumes a disproportionate amount of time to be imported, and we only need that in a pretty specific situation, so now it will be imported dynamically when required.

---

What do you think? Any ideas to optimize it more?

**Comment by igorborgest** [comment]
The idea is to publish this optimization in the version `1.10.1` next Wednesday.
Do you mind to install it directly from our repository and check before the official release? (p.s. no lambda layer available yet)

```sh
pip install git+https://github.com/awslabs/aws-data-wrangler.git --no-use-pep517
```

**Comment by jesusch** [comment]
Hi @igorborgest 

thanks for your reply. I guess I found my problem.
I was using dynamic imports at runtime, which are getting billed.
Offloading that into the Init Stage helped me avoiding this. 
Yet I have to recursive load anything :D

Thanks for your work

**Comment by igorborgest** [comment]
Released on version [1.10.1](https://github.com/awslabs/aws-data-wrangler/releases/tag/1.10.1)
